### ISSUE TITLE
我运行的时候抱错为这个  怎么办啊

### ISSUE THREAD

**Issue by YC19940619** [Hanks10100/wasm-examples#2]
WebAssembly Instantiation: table import 0 is smaller than initial 4, got 0 at fetch.then.then.then.module

**Comment by ixqbar** [comment]
```
emcc math.c -Os -s BINARYEN_ASYNC_COMPILATION=0 -s WASM=1 -s SIDE_MODULE=1 -o math.wasm

function loadWebAssembly(filename, imports = {}) {
  return fetch(filename)
    .then(response => response.arrayBuffer())
    .then(buffer => WebAssembly.compile(buffer))
    .then(module => {
      imports.env = imports.env || {}
      Object.assign(imports.env, {
        memoryBase: 0,
        tableBase: 0,
        DYNAMICTOP_PTR: 0,
        tempDoublePtr: 0,
        ABORT: 0,
        memory: new WebAssembly.Memory({ initial: 256, maximum: 256 }),
        table: new WebAssembly.Table({ initial: 2, maximum: 8, element: 'anyfunc' }),
        abortStackOverflow: function (allocSize) { },
        nullFunc_X: function () { },
        abort: function () { }
      })
      return new WebAssembly.Instance(module, imports)
    })
}

loadWebAssembly('math.wasm').then(instance => {
    const square = instance.exports._square
    console.log('2^2 =', square(2))
    console.log('3^2 =', square(3))
    console.log('(2 + 5)^2 =', square(2 + 5))
})
```

**Comment by terepiim** [comment]
what @jonnywang said is perfectly correct and some changes are needed  accroding to your question. .Just like this:
 `table: new WebAssembly.Table({ initial: 4, maximum: 8, element: 'anyfunc' })`

**Comment by jsOcean** [comment]
我运行了你代码里面的wasm文件是正确的，但是我自己生成的wasm运行会报错，我使用的命令是
emcc math.c -Os -s WASM=1 -s SIDE_MODULE=1 -o math.wasm
我的环境是
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.38.36
clang version 6.0.1 (Cswircachegitchromium.googlesource.com-external-github.com-emscripten--core-emscripten--fastcomp--clang 98df4be387dde3e3918fa5bbb5fc43e1a0e1daac) (Cswircachegitchromium.googlesource.com-external-github.com-emscripten--core-emscripten--fastcomp 1b4148f39a69c7fc62edadd85e4122b68694dfb7) (emscripten 1.38.31 : 1.38.31)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: D:\emsdk\fastcomp\fastcomp\bin
shared:INFO: (Emscripten: Running sanity checks)
求指点

**Comment by jsOcean** [comment]
这是我的报错的信息(index):30 Uncaught (in promise) LinkError: WebAssembly.Instance(): Import #0 module="env" function="__memory_base" error: global import must be a number or WebAssembly.Global object

**Comment by Hanks10100** [comment]
Fixed by dad3d379dc501c60bfed24b8846ddaf5f0ff8d63
